runtime/compiler/control/CompilationThread.cpp

```diff
@@ -6781,18 +6771,14 @@ TR::CompilationInfoPerThreadBase::installAotCachedMethod(
          // fact that AOT methods are loaded at a lower count than when they were compiled so the JVM is given less opportunity to resolve things.
          // The new hint tells us that a previous relocation failed a validation so specify a higher scount for the next run to give it more chance
          // to resolve things.
-         switch (returnCode)
+         if (reloRuntime()->isValidationError(reloErrorCode))
```

Comment:
While this now also addresses all the other validation failures, it also will store a hint for
```
methodEnterValidationFailure
methodExitValidationFailure
exceptionHookValidationFailure
stringCompressionValidationFailure
tmValidationFailure
osrValidationFailure
```
If this is not desired, I can update the code to tag these with a different relo error code type.